src/features/Chat/components/ActiveWindow/Header/Buttons.tsx

```diff
@@ -15,15 +16,23 @@ type Props = {
 const Buttons = ({ chat, showSearch }: Props) => {
   const { t } = useTranslation('chat');
 
-  const [dialogVariant, setDialogVariant] = useState<DialogVariant>('archive');
+  const [dialogVariant, setDialogVariant] =
+    useState<ConfirmationDialogVariant>('archive');
   const [isDialogOpen, setIsDialogOpen] = useState(false);
   const closeDialog = () => setIsDialogOpen(false);
 
-  const openDialog = (variant: DialogVariant) => {
-    setDialogVariant(variant);
-    setIsDialogOpen(true);
+  const openDialog = (variant: ConfirmationDialogVariant | 'report') => {
+    if (variant === 'report') {
+      setIsReportDialogOpen(true);
+    } else {
+      setDialogVariant(variant);
+      setIsDialogOpen(true);
+    }
   };
```

Comment:
This is a bit too confusing in my opinion.
Could we use different callback for opening the reportDialog and then a normalDialog?

If not, I personally despise if-else statements.
Instead i would just use early return
```javascript
if (something) return false
return true
```

Reply:
Oh yeah! I think this was some old code from when I tried to have one dialog component for all of the different variants.

Now refactored and much shorter! 💪